src/app/dashboard/Missions/MissionFundedStatusRow.jsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { Mission } from "../../model";
+import AttachMoneyIcon from "@material-ui/icons/AttachMoney";
+import { Box, Container, Grid, Paper } from "@material-ui/core";
+
+const Row = ({ children, Icon }) => {
+  if (!children) return null;
+  return (
+    <Grid container wrap="nowrap" alignItems="center">
+      <Box marginRight="5px" width="20px">
+        {Icon && <Icon color="primary" />}
+      </Box>
+      {children}
+    </Grid>
+  );
+};
+
+const MissionFundedStatusRow = ({ classes, mission }) => {
+  let missionFundedStatusText;
+  let fundedDateText = mission.fundedDate ? " on " + mission.fundedDate : "";
+  switch (mission?.fundedStatus) {
+    case Mission.FundedStatus.fundedbydonation:
+      missionFundedStatusText = "Funded By Donation" + fundedDateText;
+      break;
+    case Mission.FundedStatus.fundedbyrecipient:
+      missionFundedStatusText = "Funded By Recipient" + fundedDateText;
+      break;
+    case Mission.FundedStatus.fundingnotneeded:
+      missionFundedStatusText = "Funding Not Needed";
+      break;
+    case Mission.FundedStatus.notfunded:
+      missionFundedStatusText = "Not Yet Funded";
+      break;
+    default:
+      throw Error("mission funded status not exist", mission.fundedStatus);
```

Comment:
Is this an error that should be default?

Can a mission not have a FundedStatus?

Reply:
I don't know, I was trying to reuse code from someone else.  I think this was @mat10tng 's?

Reply:
yes, this is an error.

Reply:
Are we still fixing this?